packages/generate-persisted-query-manifest/src/index.ts

```diff
@@ -40,6 +44,16 @@ export interface PersistedQueryManifestConfig {
    */
   documents?: string | string[];
 
+  /**
+   * A `DocumentTransform` instance that will be used to transform the GraphQL
+   * document before it is saved to the manifest.
+   *
+   * For more information about document transforms, see the [Document
+   * transforms](https://www.apollographql.com/docs/react/data/document-transforms)
+   * documentation page.
+   */
+  documentTransform?: DocumentTransform;
```

Comment:
Is this the best approach or should we just have a `clientConfig: Partial<ApolloClientOptions<any>>` field?  Maybe Omitting link and cache, though we should probably also have a way to configure the cache?

Reply:
Thats not a bad idea. It would future proof it a bit more and could avoid a situation where you're using Apollo Client v3.7 where the `documentTransform` option isn't available. In that case, the client options should reflect the version you're on.

But you're right that we probably should omit a few options that don't make sense (I can also think of the `uri` option for example). Let me take a look at the full client options list and see what makes sense to omit.

Reply:
Looking a bit more through the [client options](https://github.com/apollographql/apollo-client/blob/f39392d7424d1eda6850dbd0db050a565772c30e/src/core/ApolloClient.ts#L46-L123) I'd actually say that almost all of these are unnecessary for manifest generation since a lot of them are for actual network calls. Let's talk through these in our chat later and figure out how we feel about it.

Reply:
FWIW I didn't mean to omit them because they don't make sense, but rather because they're what we explicitly pass in ourselves.

Reply:
We talked on Zoom and I agree that there aren't any other compelling options other than documentTransform so we might as well just do it this way.